### ISSUE TITLE
A few questions about transient scrollbars and some other things

### ISSUE THREAD

**Issue by JaneSmith** [tsujan/Kvantum#174]
Hello. I hope it's okay to ask a few questions in a GitHub issue like this. I did come across a post on KDE-Look.org suggesting that it was okay, so that's what I'm doing.

First of all: transient scrollbars. Wow, this is actually a thing! I've been searching for YEARS for a way to get transient scrollbars in KDE / Qt applications. I've asked so many times and nobody could ever give me an answer or a solution. I only found out about Kvantum a couple of days ago when somebody finally mentioned that the Kvantum Adapta theme could do it. This is really great and I'm happy to find this - but it seems like it's been a thing for a year or more, so it's sad that this was so hard to find.

I'm largely happy with how transient scrollbars work, but they still don't work quite as nicely as Gnome's implementation. In Gnome / GTK, scrollbars have three states (in the context of transient scrollbars): invisible, slim, and full. When you're not doing anything, the scrollbar doesn't show up. When you scroll by touch input or mouse wheel, you get a slim scrollbar to indicate the current position. When you mouse over or directly grab the scrollbar, you get the full thick scrollbar (for easier mouse usage).

**1. Kvantum seems to do the invisible and full parts, but not the in-between slim state. Is there any way to accomplish this? Am I missing something, or is it not a feature? If it isn't, could we get this as a feature? I'd really love it. It would also be nice if the "invisible" state could be optional, and you could keep a "slim" scrollbar always visible until you mouse over/grab it. Configurable, I mean.**

Aside from that, I wanted to ask a few general questions:

**2. What's the difference between Kvantum, QtCurve, and "standard" Qt themes?**

**3. Is there any performance impact to using Kvantum?**

**4. Could you consider making your documentation more discoverable?** The "landing page" for this project on GitHub only shows a one-line description, with nothing more than that. You can only find the README.md file by digging in deeper into the source files. It took me a bit of time to find this - I was originally going to ask if there even was any documentation, as I hoped to maybe port the GTK Vertex theme. Fortunately, I did manage to find this in the end. But I think it would be a lot easier if there was a README.md file in the root of the project, so that it displays on the project's home page.

Thanks in advance.

**Comment by tsujan** [comment]
> I've been searching for YEARS for a way to get transient scrollbars in KDE / Qt applications.

Me too. I even thought it should be impossible under Linux (although maybe possible under Mac) because otherwise, KDE would have it. But, at last, I realized that it was quite possible :)

Actually, I learned that I should have more doubts about what other developers might say :D

> Kvantum seems to do the invisible and ... Is there any way to accomplish this?

Yes, of course. It depends on the theme. To see an example, select and activate KvSimplicity  or KvSimplicityTurquoise with Kvantum Manager (please use the latest version of Kvantum).

> What's the difference between Kvantum, QtCurve, and "standard" Qt themes?

All of them are Qt style plugins (sometimes called Qt style engines) but of course, they're very different from each other. The standard Qt style (Fusion) is just a simple style with no variation -- except for the color scheme. The same is true for KDE styles (Oxygen and Breeze). QtCurve is more flexible. You could make very different themes for it. However, all QtCurve themes look like QtCurve ;)

Kvantum is much more flexible than QtCurve. You could activate various themes from Kvantum Manager and see how they're different from each other. For example, the default Kvantum Theme, KvBeige and KvCurves3d are so different that it may not be obvious that all of them are themes for a single style engine. That was my goal: making the most flexible style engine.

> Is there any performance impact to using Kvantum?

My work computer is a powerful laptop but I've kept my 9-year-old desktop computer for tests. Actually, I started to make Kvantum on that computer and saw no tangible impact.

Of course, when you open an app with Kvantum, it'll use a little more memory than when you open it with Fusion: the simpler (more elementary) the style plugin, the less memory -- but I wouldn't call that an "impact". Kvantum uses as much memory as its themes need -- some themes more, some less -- but it doesn't have any memory leak and doesn't waste the RAM.

> Could you consider making your documentation more discoverable?

The real doc is `Theme-Config.pdf` and `Theme-Making.pdf` inside the folder `doc` in **the source**. Packagers should put them in `/usr/share/doc/kvantum/doc` but that's out of my control. I've really tried to make the best doc possible. Believe me or not, I sometimes consult it myself ;)

Making themes for Kvantum isn't an easy task, exactly because of Kvantum's high flexibility. It takes a day for me to make a theme and the next day, I should clean it up. However, I'm sure that everyone who  consults the doc and gets his/her hands dirty, could make a nice theme based on his/her taste. That was definitely proved the first month after I made Kvantum...

Thanks for your interest in Kvantum!

**Comment by JaneSmith** [comment]
Thank you very much for your thorough and detailed response. I appreciate it!

You're absolutely right about the KvSimplicity theme doing transient scrollbars the way I wanted. I'm very happy to see that Kvantum supports that then.

Thanks for the explanation about Fusion and QtCurve. That definitely helps me to understand better how this all works. For a long time, theming on KDE / Qt has seemed either very lacking or very complicated to me, and I was starting to think Breeze was basically the only acceptable option on a Plasma desktop, while Gnome seems to have hundreds of good themes. I see that isn't the case after all. Kvantum seems wonderful and exactly what I wanted. I just wish I'd found it sooner. :)

I'll definitely dive into the Theme-Config.pdf and Theme-Making.pdf documents and see what I can come up with. Thanks a lot!

**Comment by tsujan** [comment]
You're very welcome!

> ... while Gnome seems to have hundreds of good themes.

What a similarity between your viewpoint and mine! First the transient scrollbars and now Gnome.

I migrated to the Qt world from the GTK world since Gnome-3.4. Gnome-3.6 wasn't tolerable and I felt stupid using that crap. When I came to the Qt world, I chose KDE -- there was no other choice in that time but now, we have LXQt -- and saw how bad it looked compared to Gnome. QtCurve was good but not good enough, IMO. Since I liked Qt on the programming level, I couldn't let its look remain so poor ;) After all, we make *GUI* apps. Strange that most Qt developers have such a bad taste!
